Framework/Beamline/inc/MantidBeamline/ComponentInfo.h

```diff
@@ -48,6 +48,8 @@ class MANTID_BEAMLINE_DLL ComponentInfo {
                 const std::vector<std::pair<size_t, size_t>> &ranges);
   std::vector<size_t> detectorIndices(const size_t componentIndex) const;
   size_t size() const;
+  bool operator==(const ComponentInfo& other) const;
+  bool operator!=(const ComponentInfo& other) const;
```

Comment:
I am not sure what these are needed for. Also keep in mind that it will soon not be possible to provide this anymore -- once you add positions and rotations to `ComponentInfo` exact comparison are not useful, see `DetectorInfo::isEquivalent`.